src/compiler/binder.ts

```diff
@@ -2503,8 +2503,27 @@ namespace ts {
             }
         }
 
+        function needParameterStrictModeCheck(node: Node) {
+            if (node.parent.kind === SyntaxKind.ConstructorType || node.parent.kind === SyntaxKind.FunctionType) {
+                return false;
+            }
+
+            if (node.parent.parent && (
+                node.parent.parent.kind === SyntaxKind.InterfaceDeclaration ||
+                node.parent.parent.kind === SyntaxKind.TypeLiteral
+            )) {
+                return false;
+            }
+
+            if (isDeclarationFile(file) || isInAmbientContext(node)) {

```

Comment:
isInAmbientContext actually checks the parent file at the end.

Also do not use `file` directly, you do not know what file you are checking, and what it cases the checker to jump into. no guarantees that the checker proceeds to check all nodes in a file before jumping to another.